### DISCUSSION TITLE
How to parse nested JSON from Kafka?

### DISCUSSION THREAD

**Discussion by aleksandrpugachev** [ClickHouse/ClickHouse#52942] [Q&A]
Hey there! I'm trying to parse Kafka messages by ClickHouse Kafka engine. My JSON have the next structure:
`{
	"field1": 1022127964,
	"field2": 1690793330,
	"nested_fields": {
		"unique_id": 1022127964,
		"name": "field_name",
		"nested_array": [{
			"unique_id": 3916719253,
			"name": "name",
			"value_type": 14,
			"value": "field_value"
		}, {
			"unique_id": 3916719253,
			"semanticCode": 0,
			"name": "type",
			"value_type": 14,
			"value": "field_value2"
		}, {
			"unique_id": 3916719253,
			"semanticCode": 0,
			"name": "d_tablet",
			"value_type": 14,
			"value": "field_value3"
		}],
		"flag1": false,
		"flag2": false,
		"flag3": false,
		"flag4": false
	},
	"header": {
		"unique_id": 1815404941,
		"p_Version_": 0,
		"p_Type_": 0,
		"version_": 0,
		"sourceIPv4Addr_": 0
	},
	"client_info": {
		"unique_id": 2089513335,
		"p_info_": {
			"unique_id": 2370838213,
			"version1": 0,
			"version2": 0,
			"version3": 0,
			"version4": 0,
			"hotfix": 0,
			"type": 117,
			"user_Data": null,
			"machineTypeBytes": "e5 29"
		},
		"protocol_": "HTTPS",
		"port_": 0,
		"user_data_": null
	}
}`

JSONEachRow function cannot parse this structure because of nested array with JSONs. I tried to use another JSON functions but without any luck. The functions usually return the error message 'Cannot extract table structure from JSONEachRow format file. Error: Cannot determine table structure by first 25000 rows of data, because some columns contain only Nulls'

Does anyone know how to parse JSON with the same structure?

**Comment by UnamedRus** [comment]
Easiest way is to use MV as parser, so Kafka table will have only one column which will store whole JSON message

https://kb.altinity.com/altinity-kb-schema-design/altinity-kb-jsonasstring-and-mat.-view-as-json-parser/
